dashboard/database.py

```diff
@@ -0,0 +1,135 @@
+"""Functions for interacting with the database."""
+
+from os import environ as ENV
+
+from psycopg import connect, Connection
+from dotenv import load_dotenv
+import streamlit as st
+import pandas as pd
+
+
+@st.cache_resource
+def get_connection() -> Connection:
+    """gets a connection"""
+    connection = connect(
+        port=ENV["DB_PORT"],
+        dbname=ENV["DB_NAME"],
+        host=ENV["DB_ENDPOINT"],
+        user=ENV["DB_USER"],
+        password=ENV["DB_PASSWORD"]
+    )
+    return connection
+
+
+@st.cache_resource(ttl="1hr")
```

Comment:
`st.cache_data` would be more suitable to cache functions that return data
https://docs.streamlit.io/develop/api-reference/caching-and-state/st.cache_data

Reply:
Good spot thank you! Okay I'll keep the `st.cache_resource` only for the connection function and change the others to `st.cache_data`